feat: Support using tox with `tox-uv`

mateoflorido opened the pull request:
### Overview

Add support for using tox with `uv` while running integration tests.

### Rationale

When a charm uses the `uv` plugin, certain environments use the runner environment key, which depends on the tooling available in `uv` and `tox-uv`.

For instance:
```
Run tox -e integration --notest --list-dependencies 2>&1
Traceback (most recent call last):
  File "/home/ubuntu/.local/bin/tox", line 8, in <module>
    sys.exit(run())
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/run.py", line 20, in run
    result = main(sys.argv[1:] if args is None else args)
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/run.py", line 46, in main
    return handler(state)
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/cmd/legacy.py", line [11](https://github.com/canonical/rawfile-localpv-operator/actions/runs/15475186584/job/43569031989#step:8:12)5, in legacy
    return run_sequential(state)
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/cmd/run/sequential.py", line 25, in run_sequential
    return execute(state, max_workers=1, has_spinner=False, live=True)
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/cmd/run/common.py", line [16](https://github.com/canonical/rawfile-localpv-operator/actions/runs/15475186584/job/43569031989#step:8:17)7, in execute
    state.envs.ensure_only_run_env_is_active()
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/env_select.py", line 436, in ensure_only_run_env_is_active
    envs, active = self._defined_envs, self._env_name_to_active()
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/env_select.py", line 258, in _defined_envs
    run_env = self._build_run_env(name)
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/session/env_select.py", line 3[19](https://github.com/canonical/rawfile-localpv-operator/actions/runs/15475186584/job/43569031989#step:8:20), in _build_run_env
    runner = REGISTER.runner(cast("str", env_conf["runner"]))
  File "/home/ubuntu/.local/pipx/venvs/tox/lib/python3.10/site-packages/tox/tox_env/register.py", line 72, in runner
    return self._run_envs[name]
KeyError: 'uv-venv-lock-runner'
Error: Process completed with exit code 1.
````
### Workflow Changes

<!-- Any high level changes to workflows and why -->

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The [changelog](`../docs/changelog.md`) is updated with user-relevant changes in the format of [keep a changelog v1.1.0](https://keepachangelog.com/en/1.1.0/)

<!-- Explanation for any unchecked items above -->

canonicalmateo committed:
feat: Support using tox with `tox-uv`

addyess reviewed (commented):
@mateoflorido i wonder if i spotted a bug

canonicalmateo committed:
fix: pass `python-version` to integration jobs

addyess reviewed (approved):
Nice, thanks @mateoflorido

arturo-seijas commented on .github/workflows/integration_test_run.yaml:
```diff
@@ -200,7 +222,16 @@ jobs:
           echo "UNIQUE_ARTIFACT_SUFFIX=$unique_artifact_suffix" >> $GITHUB_ENV
+      - name: Setup Astral UV
+        if: ${{ inputs.with-uv }}
+        uses: astral-sh/setup-uv@v5
```
Can you pin this action to a specific version? We aim to have repeatable builds :)

mateoflorido replied:
Sure, it's currently pinned to `v5`. Should I pin it to a specific commit or the latest version?

arturo-seijas commented on .github/workflows/workflow_test.yaml:
```diff
@@ -46,6 +46,14 @@ jobs:
       trivy-image-config: "tests/workflows/integration/test-upload-charm/trivy.yaml"
       self-hosted-runner: true
       self-hosted-runner-label: "edge"
+  integration-with-uv:
```
The `python-version` option is not tested. Can  you set it here so that that part of the code is exercised?

arturo-seijas reviewed (commented):
Thanks for your contribution! I've left a couple of comments

canonicalmateo committed:
fix: test `python-version`

mateoflorido commented:
@arturo-seijas I've addressed the comments and bumped the version of `setup-uv`. Please let me know if you have further comments. Thanks for the review!

canonicalmateo committed:
Merge branch 'main' into KU-3488/with-uv-support

canonicalmateo committed:
fix: update juju patch revision

mateoflorido commented:
@arturo-seijas @nrobinaubertin just fixed the issue with `py312` in the integration tests. It looks like one of the tests is blocked installing LXD.

arturo-seijas committed:
Merge branch 'main' into KU-3488/with-uv-support

arturo-seijas committed:
Update changelog.md

Signed-off-by: arturo-seijas <102022572+arturo-seijas@users.noreply.github.com>

arturo-seijas reviewed (commented):
Thanks for your contribution. Some of the tests are not passing because they rely on repository secrets, so PRs from forks won't pass CI. You should have permissions to push to the repository so, if you were to contribute again, please consider pushing instead of forking :)
